### PR TITLE
Gh 317 public message media not displayed

### PR DESCRIPTION
**Pull request by JAshwini**
Initially activity type and content was getting updated directly in database on `bp_activity_posted_update` action, but because of this the Ajax request was not getting updated, so added filter `bp_activity_type_before_save` and `bp_activity_content_before_save`.

### CODE REVIEW

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 1]
```diff
@@ -198,6 +200,37 @@ function bp_get_activity_parent_content( $content ) {
...
+	/**
+	 * This function will check for the media file attached to the actitvity and accordingly will set type.
+	 *
+	 * @param string $type Type of the Activity.
+	 * @return string      Filtered value of the activity type.
+	 */
+	public function bp_activity_type_before_save( $type ) {
+
+		$rtmedia_attached_files = filter_input( INPUT_POST, 'rtMedia_attached_files', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY );
+		if ( is_array( $rtmedia_attached_files ) ) {
```
Comment (abhijitrakas) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Could we use `! empty` here? check all other places.

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 2]
```diff
@@ -198,6 +200,37 @@ function bp_get_activity_parent_content( $content ) {
 		return $content;
 	}
 
+	/**
+	 * This function will check for the media file attached to the actitvity and accordingly will set type.
+	 *
+	 * @param string $type Type of the Activity.
```
Comment (abhijitrakas) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Maintain alignment after param type.

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php (hunk 1 above)
Comment (JAshwini) [reply app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Sure :+1:

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php (hunk 2 above)
Comment (JAshwini) [reply app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Done :+1:

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php (hunk 1 above)
Comment (JAshwini) [reply app/main/controllers/activity/RTMediaBuddyPressActivity.php]: Done :+1:

**Pull request merged**